apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -360,36 +377,48 @@ export const getResourcePermalinkTree = async (
   siteId: number,
   resourceId: number,
 ) => {
-  const resourcePermalinks = await db
-    .withRecursive("Ancestors", (eb) =>
-      eb
-        // Base case: Get the actual resource
-        .selectFrom("Resource")
-        .where("Resource.siteId", "=", siteId)
-        .where("Resource.id", "=", String(resourceId))
-        .select(defaultResourceSelect)
-        .unionAll((fb) =>
-          fb
-            // Recursive case: Get all the ancestors of the resource
-            .selectFrom("Resource")
-            .where("Resource.siteId", "=", siteId)
-            .innerJoin("Ancestors", "Ancestors.parentId", "Resource.id")
-            .select(defaultResourceSelect),
-        ),
-    )
-    .selectFrom("Ancestors")
-    .select("Ancestors.permalink")
-    .execute()
+  return db.transaction().execute(async (tx) => {
+    // Guard against invalid resource
+    const resource = await getById(tx, {
+      siteId,
+      resourceId,
+    }).executeTakeFirst()
+    if (!resource) {
+      return []
+    }
 
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-  return resourcePermalinks.map((r) => r.permalink).reverse() as string[]
+    const resourcePermalinks = await tx
+      .withRecursive("Ancestors", (eb) =>
+        eb
+          // Base case: Get the actual resource
+          .selectFrom("Resource")
+          .where("Resource.siteId", "=", siteId)
+          .where("Resource.id", "=", String(resourceId))
+          .select(defaultResourceSelect)
+          .unionAll((fb) =>
+            fb
+              // Recursive case: Get all the ancestors of the resource
+              .selectFrom("Resource")
+              .where("Resource.siteId", "=", siteId)
+              .innerJoin("Ancestors", "Ancestors.parentId", "Resource.id")
+              .select(defaultResourceSelect),
+          ),
+      )
+      .selectFrom("Ancestors")
+      .select("Ancestors.permalink")
+      .execute()
+
+    return resourcePermalinks.map((r) => r.permalink).reverse()
+  })
 }
 
 export const getResourceFullPermalink = async (
   siteId: number,
   resourceId: number,
 ) => {
   const permalinkTree = await getResourcePermalinkTree(siteId, resourceId)
-
+  if (permalinkTree.length === 0) {
+    return null
+  }
```

Comment:
just a sanity check to ensure that this `null` value is handled properly on frontend